demo-shell/src/app/app.module.ts

```diff
@@ -189,24 +187,17 @@ registerLocaleData(localeSv);
         CloudFiltersDemoComponent,
         TemplateDemoComponent,
         PeopleGroupCloudDemoComponent,
-        CloudSettingsComponent,
         ConfirmDialogExampleComponent,
         FormCloudDemoComponent,
-        ConfirmDialogExampleComponent,
-        CommunityCloudComponent,
-        CommunityTasksCloudDemoComponent,
-        CommunityCloudFiltersDemoComponent,
-        CommunityProcessesCloudDemoComponent,
-        CommunityStartProcessCloudDemoComponent,
-        CommunityStartTaskCloudDemoComponent,
-        CommunityProcessDetailsCloudDemoComponent,
-        CommunityTaskDetailsCloudDemoComponent
+        ConfirmDialogExampleComponent
     ],
     providers: [
         {
             provide: HTTP_INTERCEPTORS, useClass:
             AuthBearerInterceptor, multi: true
         },
+        { provide: PROCESS_FILTERS_SERVICE_TOKEN, useClass: UserPreferenceCloudService },
```

Comment:
Why having tokens instead of replacing the service?

Reply:
I think tokens are better than using actual services to be rewritten. Why would you need to import the whole service, when you just need to have an identifier for the IoC. Unfortunately Angular's DI can't do this with interfaces, that is why I think tokens are inevitable.
I don't think either, it is that complex, that developers couldn't understand. Almost all of the Angular related components use tokens.

Reply:
I agree it is always better to have an interface in the code instead of a concrete class that can be overridden from the outside. The idea is that by default the Cloud module will inject the localstorage implementation and in case of enterprise we will inject the UserPreferenceService